arelle/plugin/validate/ESEF/ESEF_Current/ValidateXbrlFinally.py

```diff
@@ -690,7 +700,10 @@ def checkFootnote(elt: ModelInlineFootnote | ModelResource, text: str) -> None:
                 if esefDisclosureSystemYear >= 2024:
                     if not f.id:
                         factsMissingId.append(f)
-                    if isinstance(f, ModelInlineFact) and f.concept is not None and f.isEscaped != f.concept.isTextBlock:
+                    if isinstance(f, ModelInlineFact) and f.concept is not None and (
+                        (esefDisclosureSystemYear == 2024 and f.isEscaped != f.concept.isTextBlock) or
+                        (esefDisclosureSystemYear >= 2025 and (not f.isEscaped and 
+                            (f.concept.isTextBlock or (f.value and escapeWorthyStr.match(f.value)))))):
                         modelXbrl.error("ESEF.2.2.7.improperApplicationOfEscapeAttribute",
```

Comment:
There's a lot going on in this single statement which can make it difficult to read and maintain.

Suggest splitting into separate statements:
```python
if isinstance(f, ModelInlineFact) and f.concept is not None:
    if esefDisclosureSystemYear == 2024:
      hasEscapeIssue = f.isEscaped != f.concept.isTextBlock
    elif esefDisclosureSystemYear >= 2025:
      shouldBeEscaped = (
          f.concept.isTextBlock or
          (f.value and escapeWorthyStr.match(f.value))
      )
      hasEscapeIssue = not f.isEscaped and shouldBeEscaped

    if hasEscapeIssue:
        modelXbrl.error("ESEF.2.2.7.improperApplicationOfEscapeAttribute",
```